lib/time.py

```diff
@@ -23,3 +55,37 @@ def sleep(secs):
 
 def time():
   return float(Now().UnixNano()) / Second
+
+
+_strftime_map = {
+    'Y': '2006', 'y': '06', 'b': 'Jan', 'B': 'January', 'm': '01', 'd': '02',
+    'a': 'Mon', 'A': 'Monday',
+    'H': '15', 'I': '03', 'M': '04', 'S': '05', 'L': '.000',
+    'p': 'PM',
+    'Z':  'MST', #'z': '-0700',
+    '%': '%',
+    # TODO: Implement c, x, X, j, U, w, W
+}
+
+def strftime(layout, tt=None): #pylint: disable=missing-docstring
+  ret = ''
+  t = (Unix(int(mktime(tt)), 0) if tt else Now()).Local()
+  prev, n = 0, layout.find('%')
+  while 0 <= n <= len(layout) - 2:
```

Comment:
I like the overall approach of this function. I just have a few comments about specific details that are all kind of related:

* The way the loop condition works right now, there are two reasons why it might exit (% was found at the end of the string or % was not found.) I think it'd be easier to understand if there was one condition: `n != -1`
* It's cheaper to append the pieces of the result to a list and join them at the end.
* It seems like strftime discards the % for escapes that it doesn't recognize so I think you can remove the TODO:

```
>>> time.strftime('%3')
'3'
```

* The logic is a little simpler if you use _strftime_map.get() instead of checking `next_val in _strftime_map`.

I've written an example implementation (just to prove to myself that it's simpler, no need to copy it verbatim) that demonstrates these ideas:

```
ret = []
prev, n = 0, layout.find('%', 0, -1)
while n != -1:
  ret.append(layout[prev:n])
  c = _strftime_map.get(layout[n+1])
  if c:
    ret.append(t.Format(c))
  else:
    ret.append(c)
  prev, n = n, layout.find('%', n + 2, -1)
ret.append(layout[prev:])
return ''.join(ret)
```